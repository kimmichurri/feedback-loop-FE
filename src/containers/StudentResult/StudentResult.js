import React, { Component } from 'react'
import StudentResult from '../StudentResult/StudentResult'
import { connect } from 'react-redux'
import { handleGet } from '../../thunks/handleGet'
import cogoToast from 'cogo-toast'

export class StudentResult extends Component {
  constructor(props) {
    super(props);
    this.state = {
      studentResult: [],
      classResult: [],
    }
  }

  componentDidMount = async () => {
    await this.fetchStudentResult()
    await this.fetchClassResult()
  }

  fetchStudentResult = () => {
    const myKey = await localStorage.getItem('currentUser')
    const url = `https://turing-feedback-api.herokuapp.com/api/v1/surveys/${this.props.survey.id}/averages/student?api_key=${myKey}`
    const studentResult = await this.props.handleGet(url)

    this.setState({
      surveyResult: studentResult
    })
  }

  fetchClassResult = () => {
    const url = `https://turing-feedback-api.herokuapp.com/api/v1/surveys/${this.props.survey.id}/averages`
    const classResult = await this.props.handleGet(url)

    this.setState({
      classResult: classResult
    })
  }

  displayQuestionResult = (question) => {
    var question = question
    return(
      <div className='result-question'>
        Question: {question.description}
        Class Average: {this.state.classResult}
        Your Average: {this.state.studentResult}
      </div>
    )
  }

  render() {
    return(
      <div className='student-result'>
        <p className='response-survey-name'>{this.state.surveyName}</p>
        {this.state.studentResults.length &&
          this.props.survey.questions.map(question => {
            this.displayQuestionResult(question)
          })
        }
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  studentSurveys: state.studentSurveys,
  user: state.user
})

export const mapDispatchToProps = (dispatch) => ({
  handlePost: (url, options) => dispatch(handlePost(url, options)),
  handleGet: (url) => dispatch(handleGet(url)),
  setStudentSurveys: (surveys) => dispatch(setStudentSurveys(surveys))
})

export default connect(mapStateToProps, mapDispatchToProps)(StudentResult)
