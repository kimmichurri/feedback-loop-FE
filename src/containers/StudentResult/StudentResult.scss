.result-card {
  -webkit-box-shadow: 0 0 2px 1px rgba(0,0,0,.2);
  box-shadow: 0 0 2px 1px rgba(0,0,0,.2);
  color: $accent-charcoal;
  cursor: pointer;
  padding: 1rem;
  background-color: white;
  width: 95%;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.result-data p {
  text-align: left;
}
