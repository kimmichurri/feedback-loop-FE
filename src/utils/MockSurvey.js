var survey = {
    "created_at": "2019-05-26T17:18:25",
    "groups": [
        {
            "members": [
                {
                    "cohort": "1811",
                    "id": 3,
                    "name": "Scott Thomas",
                    "program": "B",
                    "status": "Active"
                },
                {
                    "cohort": "1811",
                    "id": 8,
                    "name": "Jessica Hansen",
                    "program": "F",
                    "status": "Active"
                },
                {
                    "cohort": "1811",
                    "id": 2,
                    "name": "April Dagonese",
                    "program": "B",
                    "status": "Active"
                }
            ],
            "name": "1811"
        }
    ],
    "id": 21,
    "questions": [
        {
            "id": 19,
            "options": [
                {
                    "description": "4",
                    "id": 76,
                    "pointValue": 4
                },
                {
                    "description": "3",
                    "id": 75,
                    "pointValue": 3
                },
                {
                    "description": "2",
                    "id": 74,
                    "pointValue": 2
                },
                {
                    "description": "1",
                    "id": 73,
                    "pointValue": 1
                }
            ],
            "questionTitle": "How good is this jabroni?"
        }
    ],
    "status": "Closed",
    "surveyExpiration": "2019-05-26T17:17:42",
    "surveyName": "Inst Test",
    "updated_at": "2019-05-27T21:40:28"
}

export default survey
