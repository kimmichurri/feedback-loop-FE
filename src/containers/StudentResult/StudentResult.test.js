import React from 'react'
import { shallow } from 'enzyme'
import { StudentResult, mapStateToProps, mapDispatchToProps } from './StudentResult'
import MockSurvey from '../../utils/MockSurvey'
jest.mock('../../thunks/handleGet')

describe('StudentResult', () => {
  let mockState
  let wrapper
  let mockEvent
  let mockFn = jest.fn()

  beforeEach(() => {
    mockState = {
      studentResult: null,
      classResult: null,
      dataDisplay: "none"
    }
    wrapper = shallow(
      <StudentResult
        survey={MockSurvey}
      />
    )
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should have proper default state', () => {
    expect(wrapper.state()).toEqual({
      studentResult: null,
      classResult: null,
      dataDisplay: "none"
    })
  })

  it('should set toggle the data display', () => {
    wrapper.instance().toggleData()

    expect(wrapper.state()).toEqual({
      studentResult: null,
      classResult: null,
      dataDisplay: "flex"
    })
  })

})
