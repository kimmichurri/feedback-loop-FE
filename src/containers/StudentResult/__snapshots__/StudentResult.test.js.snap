// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudentResult should match the snapshot 1`] = `
<div
  className="student-result"
>
  <div
    className="result-card"
    onClick={[Function]}
  >
    <div
      className="result-title"
    >
      Inst Test
    </div>
    <div
      className="result-data"
      style={
        Object {
          "display": "none",
        }
      }
    >
      <div
        className="data-container"
      />
    </div>
  </div>
</div>
`;
